### ISSUE TITLE
Universal Bounding Box Regression and Its Applications

### ISSUE THREAD

**Issue by howardyclo** [howardyclo/papernotes#53]
### Metadata
- Authors: Seungkwan Lee, Suha Kwak, and Minsu Cho
- Organization: Dept. of Computer Science and Engineering, POSTECH, Korea.
- Conference: ACCV 2018
- Paper: https://arxiv.org/pdf/1904.06805.pdf

### TL;DR
![](https://github.com/howardyclo/papernotes/blob/master/images/UBBR/1.png?raw=true)
- Propose a class-agnostic (transferable to **unseen classes**) and anchor-free box regressor, *Universal Bounding-Box Regressor* (UBBR).
- UBBR takes an image and any arbitrary bounding boxes, and refines the boxes so that they enclose their nearest objects tightly, regardless of their classes.
- UBBR can also be used to generate object box proposals. Given boxes uniformly and densely sampled from image space, UBBR transforms them to approximate the boxes of their nearest objects.
- Trained on small data and perform well on unseen classes on weakly supervised object detection, object proposals, and object discovery.

### Problem
- In weakly supervised object detection, box annotations for target object classes are not given, only image-level class labels. As as result, **object bounding boxes tend to be badly localized due to the limited supervision** and **only discriminative part** is covered.

### Method
- Data: Any existing object detection dataset with image and its ground-truth boxes.
- Train: Randomly generate **roughly localized** and **diverse** boxes: Apply random transform to ground-truth boxes and its IoU should be greater than a threshold.
- Loss: Use IoU loss between ground-truth box and random box, instead of conventional L2 and smooth L1 losses (IoU loss is better). **Add tiny constant to loss to enable numerical stability**
![](https://github.com/howardyclo/papernotes/blob/master/images/UBBR/2.png?raw=true)
- **b**: input box; **UBBR(b)**: offsets predicted by UBBR; **f(.)**: refine **b** with predicted offsets.

### Experiment
- Dataset: Tested on VOC; trained on COCO-17 (remove VOC classes: 80-20=60).
- Use ImageNet-pretrained ResNet101. (Doesn't ImageNet containing "unseen" objects make experiments invalid?)
- Baselines:
    - Weakly supervised object detection:
        - [OICR](https://arxiv.org/abs/1704.00138) Iteratively refines localization.
    - Object proposals: Out-performed previous 13 methods. **Note that unlike many other methods (except SelectiveSearch [18]), UBBR does not use any images from VOC object classes for training.**
    - Object discovery:
        - Unsupervised object discovery and localization in the wild: Part-based matching with bottom-up region proposals.
        - Image co-localization by mimicking a good detectors confidence score distribution.

### Limitation
![](https://github.com/howardyclo/papernotes/blob/master/images/UBBR/3.png?raw=true)
- UBBR applied on OICR outputs may not be able to localize nested objects since UBBR is class-agnostic.

**Comment by howardyclo** [comment]
### TL;DR
![](https://github.com/howardyclo/papernotes/blob/master/images/UBBR/1.png?raw=true)
- Propose a class-agnostic (transferable to **unseen classes**) and anchor-free box regressor, *Universal Bounding-Box Regressor* (UBBR).
- UBBR takes an image and any arbitrary bounding boxes, and refines the boxes so that they enclose their nearest objects tightly, regardless of their classes.
- UBBR can also be used to generate object box proposals. Given boxes uniformly and densely sampled from image space, UBBR transforms them to approximate the boxes of their nearest objects.
- Trained on small data and perform well on unseen classes on weakly supervised object detection, object proposals, and object discovery.

### Problem
- In weakly supervised object detection, box annotations for target object classes are not given, only image-level class labels. As as result, **object bounding boxes tend to be badly localized due to the limited supervision** and **only discriminative part** is covered.

### Method
- Data: Any existing object detection dataset with image and its ground-truth boxes.
- Train: Randomly generate **roughly localized** and **diverse** boxes: Apply random transform to ground-truth boxes and its IoU should be greater than a threshold.
- Loss: Use IoU loss between ground-truth box and random box, instead of conventional L2 and smooth L1 losses (IoU loss is better). **Add tiny constant to loss to enable numerical stability**
![](https://github.com/howardyclo/papernotes/blob/master/images/UBBR/2.png?raw=true)
- **b**: input box; **UBBR(b)**: offsets predicted by UBBR; **f(.)**: refine **b** with predicted offsets.

### Experiment
- Dataset: Tested on VOC; trained on COCO-17 (remove VOC classes: 80-20=60).
- Use ImageNet-pretrained ResNet101. (Doesn't ImageNet containing "unseen" objects make experiments invalid?)
- Baselines:
    - Weakly supervised object detection:
        - [OICR](https://arxiv.org/abs/1704.00138) Iteratively refines localization.
    - Object proposals: Out-performed previous 13 methods. **Note that unlike many other methods (except SelectiveSearch [18]), UBBR does not use any images from VOC object classes for training.**
    - Object discovery:
        - Unsupervised object discovery and localization in the wild: Part-based matching with bottom-up region proposals.
        - Image co-localization by mimicking a good detectors confidence score distribution.

### Limitation
![](https://github.com/howardyclo/papernotes/blob/master/images/UBBR/3.png?raw=true)
- UBBR applied on OICR outputs may not be able to localize nested objects since UBBR is class-agnostic.
